### PR TITLE
Makes storage implementation swappable

### PR DESCRIPTION
**Pull request by gabzim**
Motivation: To most integrators, this won't make a difference. They'll be using the `StandardRelayerApp` which hides everything behind the same interface. However, it opens the door to anybody that wants to leverage their own storage solution (SQS, Postgres, etc). And it's on them to then implement middleware that uses their same db if they want to avoid having to run multiple databases.

The change is to avoid locking people to redis, even if that's what we use and recommend using publicly. We will not provide an alternative storage backend for now.

### CODE REVIEW

File: relayer/application.ts [hunk 1]
```diff
@@ -488,3 +463,6 @@ export type ContractFilter = {
   emitterAddress: string; // Emitter contract address to filter for
   chainId: ChainId; // Wormhole ChainID to filter for
 };
+
+/**
```
Comment (JoeHowarth) [inline relayer/application.ts:467]: typo?

File: relayer/application-standard.ts [hunk 2]
```diff
@@ -55,6 +59,7 @@ export type StandardRelayerContext = LoggingContext &
 export class StandardRelayerApp<
```
Comment (JoeHowarth) [inline relayer/application-standard.ts:59]: This class is starting to become a bit large s.t. if someone wants to change one of these defaults (e.g. not used missedVaa middleware) they need to copy a bunch of code. 

What about creating a `RedisRelayerApp` that is like the previous `RelayerApp` but configured with redis?

**Review by JoeHowarth — APPROVED** [review]

File: relayer/application-standard.ts (hunk 2 above)
Comment (gabzim) [reply relayer/application-standard.ts:59]: I'd like to discuss a few things before doing that but I like it. Can we tackle that in a separate pr?

Comment (JoeHowarth) [reply relayer/application-standard.ts:59]: Yeah agreed on separate pr 👍 
Happy to chat tomorrow

**Pull request merged**